NetKet/Graph/abstract_graph.hpp

```diff
@@ -15,9 +15,27 @@
 #ifndef NETKET_ABSTRACTGRAPH_HPP
 #define NETKET_ABSTRACTGRAPH_HPP
 
+#include <array>
+#include <unordered_map>
 #include <vector>
 
+// Special hash functor for the EdgeColors unordered_map
+// Same as hash_combine from boost
+namespace std {
```

Comment:
Sorry for being bold, but `namespace std` here is a bad idea. It's even considered undefined behavior by the C++ standard. I'd suggest to simply define the hash function inside the NetKet's namespace.

Reply:
Indeed, extending `std` is undefined behaviour except for a few cases. See e.g. https://en.cppreference.com/w/cpp/language/extending_std.

Reply:
Thanks, that's good to know, I will move it to `namespace netket`.